src/pid_namespace.c

```diff
@@ -0,0 +1,15 @@
+#include "jemalloc/internal/pid_namespace_externs.h"
+
+#ifdef JEMALLOC_PID_NAMESPACE
+#include <sys/stat.h>
+
+static const char* PID_NAMESPACE_PATH = "/proc/self/ns/pid";
+
+ssize_t pid_namespace() {
+  struct stat statbuf;
+  if (stat(PID_NAMESPACE_PATH, &statbuf) == -1) {
```

Comment:
Maybe try an alternative way of reading the namespace (will readlink / readlinkat work)? One concern is that if some implementation of `stat` can allocate, then it would form a circular dependency (malloc->prof_dump->stat->malloc). While we already rely on readlink not allocating: https://github.com/jemalloc/jemalloc/blob/e4817c8d89a2a413e835c4adeab5c5c4412f9235/src/jemalloc.c#L997

Reply:
Let me double check that, good callout!